wireapp/kalium
# feat: discover legal hold when sending message [WPB-5999]

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

We need to discover missing/deleted legal hold clients and changes in legal hold for conversation on Android when sending messages and handle this change properly by showing system messages.

### Solutions

Implement new function inside `LegalHoldHandler` to handle legal hold with the response after sending a message.
Create legal hold related messages with the date-time equal to the date-time of the message that's being sent/received minus 1 millisecond, to make it so that it's always shown before that message.
Fetch the info about the missing clients, at least this part of metadata that allows the app to specify the type of the client, if it's a legal hold or not.
Emit new type of failure `LegalHoldEnabledForConversationFailure` specific to sending a message to be able to ask the user if he/she still wants to send a message now after the confersation has become a subject of a legal hold, message sending is retried automatically after handling missing/deleted clients just like it has been working until now if legal hold status is not changed for that conversation, so it works more like explained in the [documentation](https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/157288246/Use+case+Sending+a+message+and+discovering+missing+deleted+legal+hold+devices).

### Testing

#### Test Coverage (Optional)


#### How to Test

Enable legal hold for some other member before logging in and syncing, then login - at this point the app doesn't yet know that a conversation or member is under legal hold, but after sending first message, it should be handled properly.

### Attachments (Optional)

https://github.com/wireapp/kalium/assets/30429749/3bf93acf-7ba9-4c6c-bb10-8306862045e1

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
